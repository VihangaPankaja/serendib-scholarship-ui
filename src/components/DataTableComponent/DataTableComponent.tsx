import React, { Component } from 'react';
import DataGrid from 'react-data-grid';

interface DataTableProps {
    columns: columnType[],
    rows: rowType[]
}

type columnType = {
    key: string;
    name: string;
};

type rowType = {
    id: number;
    name: string;
};
  
export default class DataTableComponent extends Component<DataTableProps> {
    constructor(props: DataTableProps) {
        super(props);
    }

    render(): React.ReactNode {
        return <DataGrid style={{height: '100vh'}} columns={this.props.columns} rows={this.props.rows} />;
    }
}