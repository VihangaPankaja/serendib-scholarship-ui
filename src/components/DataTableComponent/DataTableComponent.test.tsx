import { render } from "@testing-library/react";
import React from "react";
import { describe, it, expect } from "@jest/globals";
import DataTableComponent from "./DataTableComponent";

describe("Card", () => {
  it("renders", () => {
    const wrapper = render(<DataTableComponent columns={[]} rows={[]} />);
    expect(wrapper.container).toMatchSnapshot();
  });
});