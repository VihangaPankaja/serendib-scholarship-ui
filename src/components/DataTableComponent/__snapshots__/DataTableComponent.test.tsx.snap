// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders 1`] = `
<div>
  <div
    aria-colcount="0"
    aria-rowcount="1"
    class="rdg r104f42s700-beta15"
    dir="ltr"
    role="grid"
    style="height: 100vh; grid-template-rows: 35px; --rdg-header-row-height: 35px; --rdg-summary-row-height: 35px; --rdg-sign: 1;"
  >
    <div
      aria-rowindex="1"
      class="rdg-header-row h197vzie700-beta15"
      role="row"
      style="--rdg-grid-row-start: 1;"
    />
  </div>
</div>
`;
